src/python/dxpy/__init__.py

```diff
@@ -609,6 +610,12 @@ def ensure_ascii(i):
 
                 response = pool_manager.request(_method, encoded_url, headers=_headers, body=body,
                                                 timeout=timeout, retries=False, **kwargs)
+                # Handle redirection manually for symlink files
+                if response.status in range(300, 399):
+                    redirect_url = response.headers.get('Location')
+                    if try_index > 0:
+                        logger.info("[%s] %s %s: Recovered after %d retries", time.ctime(), method, _url, try_index)
```

Comment:
```suggestion
```

Reply:
this should not be longer necessary as `finally` is processed before new request